test/javascripts/turbolinks_test.coffee

```diff
@@ -98,113 +127,161 @@ describe 'Turbolinks', ->
+        handler()
+        assert handler.calledOnce
 
-        assert your_callback.calledOnce
+      @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
 
-      it 'script tags are evaluated when they are the subject of a partial replace', ->
-        window.globalStub = stub()
-        @server.respondWith([200, { "Content-Type": "text/html" }, script_response]);
+      Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
+      @server.respond()
+
+      $(document).off 'page:before-partial-replace'
+
+    it 'replaces and passes through the outermost nodes if a series of nodes got replaced', ->
+      currentBody = """
+        <div id="div1" refresh="div1">
+          <div id="div2" refresh-always>Refresh-always</div>
+        </div>
+      """
+
+      $("body").append($(currentBody))
+
+      $(document).on 'page:before-partial-replace', (ev) ->
+        nodes = ev.originalEvent.data
+        assert.equal 2, nodes.length
+        assert.equal 'div2', nodes[0].id
+        assert.equal 'div1', nodes[1].id
+
+      $(document).on 'page:load', (ev) ->
+        nodes = ev.originalEvent.data
+        assert.equal 1, nodes.length
+        assert.equal 'div1', nodes[0].id
+
+      @server.respondWith([200, { "Content-Type": "text/html" }, response_with_refresh_always])
+
+      Page.refresh(onlyKeys: ['div1'])
+      @server.respond()
 
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-        assert globalStub.calledOnce
+      $(document).off 'page:before-partial-replace'
+      $(document).off 'page:load'
 
-      it 'script tags are not evaluated if they have [data-turbolinks-eval="false"]', ->
-        window.globalStub = stub()
-        @server.respondWith([200, { "Content-Type": "text/html" }, script_response_turbolinks_eval_false]);
+      $("#div1").remove()
 
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-        assert.equal 0, globalStub.callCount
+      it 'does not trigger a full refresh when a new script is in the head', ->
+    @server.respondWith([200, { "Content-Type": "text/html" }, html_with_script_in_head])
 
-      it 'triggers the page:load event with a list of nodes that are new (freshly replaced)', ->
-        $(document).one 'page:load', (event) ->
-          ev = event.originalEvent
-          assert.equal true, ev.data instanceof Array
-          assert.equal 1, ev.data.length
-          node = ev.data[0]
+    Turbolinks.visit "/some_request"
+    @server.respond()
 
-          assert.equal "turbo-area", node.id
-          assert.equal "turbo-area", node.getAttribute('refresh')
+    assert.equal true, @pushStateStub.called, 'Should call pushState instead of redirecting.'
 
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
+  it 'inserts script with a new src from new doc into head on navigation', ->
```

Comment:
This test is terrible. It's mutating the document the test itself is running on and basically explodes teaspoon's browser side test runner if you try to view it. It's also order dependant for the same reason. 

That said currently due to the order of the tests the state of the document is such that there are no script tags in head because the previous tests provide the `html_one` string literal as a spoofed server response body. So yes, it's accurate. 

*pukes*